### PR TITLE
add device filter

### PR DESCRIPTION
**Pull request by okhiroyuki**
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red-nodes/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the mailing list/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [x] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: social/pushbullet/57-pushbullet.js [hunk 1]
```diff
@@ -511,14 +511,24 @@ module.exports = function(RED) {
...
             res.send("[]");
         }
     });
 
+    function filterdActiveDevices(devices){
+        var activeDevices = [];
+        for(var i=0;i<devices.length;i++){
+            if(chans.devices[i].active){
+                activeDevices.push(devices[i]);
+            };
```
Comment (dceejay) [inline social/pushbullet/57-pushbullet.js]: sadly we have the jsHint rules set fairly high so this ; triggers it. It can be removed.

**Comment by okhiroyuki** [comment]
This Pull Request has the following discussions.
https://discourse.nodered.org/t/hide-inactive-devices-for-node-red-node-pushbullet/11548

File: social/pushbullet/57-pushbullet.js (hunk 1 above)
Comment (okhiroyuki) [reply social/pushbullet/57-pushbullet.js]: sorry.
fix it.

Comment (dceejay) [reply social/pushbullet/57-pushbullet.js]: no problem - and you also need to accept/sign the CLA. Thanks

**Comment by okhiroyuki** [comment]
@jsf-clabot 
I signed the CLA.

**Comment by okhiroyuki** [comment]
@dceejay 
I signed the CLA.

**Comment by dceejay** [comment]
great - as soon as you update the PR with the fix it should retest/recheck.

**Comment by okhiroyuki** [comment]
Thank you.
I will test it.

File: social/pushbullet/57-pushbullet.js [hunk 2]
```diff
@@ -511,14 +511,24 @@ module.exports = function(RED) {
...
         }
         else {
             res.send("[]");
         }
     });
 
+    function filterdActiveDevices(devices){
+        var activeDevices = [];
+        for(var i=0;i<devices.length;i++){
+            if(chans.devices[i].active){
```
Comment (dceejay) [inline social/pushbullet/57-pushbullet.js]: you call this function with a parameter of `chans.devices` and define the parameter of the function as filterdActiveDevices(devices) so inside (line 525) you only need `if (devices[i].active) {`

Is that correct ?

Comment (okhiroyuki) [reply social/pushbullet/57-pushbullet.js]: I'm mistake.
I will fix it.

Comment (okhiroyuki) [reply social/pushbullet/57-pushbullet.js]: @dceejay 
fix it.

**Comment by dceejay** [comment]
Thanks for the Pull request

**Pull request merged**